tensorly/tenalg/einsum_tenalg/caching.py

```diff
@@ -0,0 +1,20 @@
+EINSUM_PATH_CACHE = dict()
+
+
+def einsum_path_cached(fun):
+    def wrapped(key, *args, **kwargs):
+        name = fun.__name__
+        try:
+            cache = EINSUM_PATH_CACHE[name]
+        except KeyError:
+            EINSUM_PATH_CACHE[name] = dict()
+            cache = EINSUM_PATH_CACHE[name]
+        try:
+            equation = cache[key]
+        except KeyError:
+            equation = fun(*args, **kwargs)
+            cache[key] = equation
```

Comment:
Is there a reason why you use `try`-`except` instead of checking if the key is in the cache beforehand? Is that more efficient?

Reply:
Yes, getting an item is just O(1) while checking for existence is O(n) and the overhead of the try/except is minimal and only happens at the first call so the overall cost is just O(1).